### PR TITLE
[Fix](predicate pushdown) Common expression not acting on any slot should not be pushed down

### LINKED ISSUE #25900: [Bug] Wrong result when using random() as where condition.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

mater

### What's Wrong?

mysql> select count(l_receiptdate) from lineitem where random() = 0;
+----------------------+
| count(l_receiptdate) |
+----------------------+
|            600037902 |
+----------------------+
1 row in set (0.48 sec)

### What You Expected?

result should be empty since random() will never be zero.

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
Issue Number: close [#25900 ](https://github.com/apache/doris/issues/25900)

If conjunc does not act an any slot, do not push it down.

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/exec/scan/vscan_node.cpp [hunk 1]
```diff
@@ -410,26 +411,48 @@ Status VScanNode::_normalize_conjuncts() {
         }
     }
 
+    std::function<bool(const VExprSPtr&)> _conjunct_is_acting_on_a_slot =
```
Comment (yiguolei) [inline be/src/vec/exec/scan/vscan_node.cpp:414]: 这种的就不要写lambda了，直接定义一个函数吧

File: be/src/vec/exec/scan/vscan_node.cpp [hunk 2]
```diff
@@ -410,26 +411,48 @@ Status VScanNode::_normalize_conjuncts() {
...
+            [&_conjunct_is_acting_on_a_slot](const VExprSPtr& expr) -> bool {
+        const auto& children = expr->children();
+        const size_t children_size = children.size();
+
+        for (size_t i = 0; i < children_size; ++i) {
+            // If any child expr does not act on a column slot
+            // return false immediately.
+            if (!_conjunct_is_acting_on_a_slot(children[i])) {
```
Comment (yiguolei) [inline be/src/vec/exec/scan/vscan_node.cpp:422]: 这样做应该是不对的，应该是，只要有一个acting on slot，那么就应该是true；其他的是false

File: regression-test/suites/correctness/test_pushdown_common_expr.groovy [hunk 3]
```diff
@@ -65,6 +65,38 @@ suite("test_pushdown_common_expr") {
...
 
+    // One conjunct. No push down.
+    order_qt_5 """
+        SELECT * FROM t_pushdown_common_expr WHERE random() > 1 OR uuid_numeric() = 'A' OR c1 = 256
+    """
+
+    // Two conjuncts.
+    // random() > 1 is executed by scan node, c1 > 0 is pushed down and executed by segment iterator
+    order_qt_6 """
+        SELECT * FROM t_pushdown_common_expr WHERE random() > 1 AND c1 > 0
```
Comment (yiguolei) [inline regression-test/suites/correctness/test_pushdown_common_expr.groovy:76]: 寄一个select * from table where random() > 1 的测试case吧

File: be/src/vec/exec/scan/vscan_node.cpp (hunk 1 above)
Comment (zhiqiang-hhhh) [reply be/src/vec/exec/scan/vscan_node.cpp:414]: 那这应该定一一个 `static bool VExpr::is_acting_on_a_slot() `吧？

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**